tests/integration/controller/mandatorymodule/deletion/mandatory_modules_deletion_controller_test.go

```diff
@@ -0,0 +1,158 @@
+package mandatory_test
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	compdescv2 "github.com/open-component-model/ocm/pkg/contexts/ocm/compdesc/versions/v2"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var ErrNoMandatoryManifest = errors.New("manifest for mandatory Module not found")
+
+const (
+	mandatoryChannel = "dummychannel"
+)
+
+var _ = Describe("Mandatory Module Deletion", Ordered, func() {
+	Context("Given Kyma with one mandatory Module Manifest CR on Control-Plane", func() {
+		kyma := &v1beta2.Kyma{}
+		It("When the Kyma CR is created", func() {
+			kyma = NewTestKyma("no-module-kyma")
+			registerControlPlaneLifecycleForKyma(kyma)
+		})
+		It("Then Kyma CR should result in a ready state and mandatory manifest is created", func() {
+			Eventually(KymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+				Should(Succeed())
+			Eventually(ManifestExistsWithAnnotation).
```

Comment:
I am not sure if it's tested anywhere...
Mandatory Module Manifests do have this label: "operator.kyma-project.io/mandatory-module:true".
Maybe we should test if this label exists here?

Reply:
Edit: I think it's the wrong place. It should be rather tested in the e2e tests.

Reply:
I will add it to the ManifestExists function, it will be called `MandatoryManifestExistsWithLabelAndAnnotation`. Since we do not know the manifest CR name, we can check if it exists filtering the label and annotation and thus directly check if those are correclty set